### PR TITLE
Support for Julia 0.4 by re-exporting Base objects

### PR DESCRIPTION
**Pull request by nalimilan**
The package needs to work on 0.4 so that packages depending on it
can support it (conditional dependencies do not exist yet).

### CODE REVIEW

**Comment by nalimilan** [comment]
BTW, could you give me permissions on JuliaArchive? I'd like to enable CI on 0.4, which for some reason doesn't happen despite the contents of configuration files.

File: test/runtests.jl [hunk 1]
```diff
@@ -19,7 +20,11 @@ badstring32  = UInt32['a']
 # Unicode errors
 let io = IOBuffer()
     show(io, UnicodeError(UTF_ERR_SHORT, 1, 10))
-    check = "UnicodeError: invalid UTF-8 sequence starting at index 1 (0xa missing one or more continuation bytes)"
+    if VERSION >= v"0.5.0-"
```
Comment (tkelman) [inline test/runtests.jl]: be more specific?

**Comment by tkelman** [comment]
> could you give me permissions on JuliaArchive?

done

File: test/runtests.jl (hunk 1 above)
Comment (nalimilan) [inline test/runtests.jl]: Done.

**Comment by nalimilan** [comment]
Thanks. Though I've been able to set up AppVeyor at https://ci.appveyor.com/project/nalimilan/legacystrings-jl, I can't find how to make it appear in the PR...

**Comment by tkelman** [comment]
Strange. It looks like it's running but not sending its status back to github?

**Comment by nalimilan** [comment]
Yeah. What do one needs to do in general apart from adding a webhook?

**Comment by tkelman** [comment]
Usually I think that should be it? We could maybe ping Feodor.

**Comment by nalimilan** [comment]
Got it, you need to revoke AppVeyor permissions and add them back for it to take into account new organization memberships: http://stackoverflow.com/questions/32607885/appveyor-doesnt-update-github-pull-request-with-the-build-status

**Comment by nalimilan** [comment]
Yay!

**Comment by tkelman** [comment]
github really had to send 10 copies of that email? :frowning: 

PR LGTM

File: src/LegacyStrings.jl [hunk 2]
```diff
@@ -40,40 +40,43 @@ import Base:
...
+            data::Array{UInt8,1}
+        end
 
-immutable UTF16String <: AbstractString
-    data::Vector{UInt16} # includes 16-bit NULL termination after string chars
-    function UTF16String(data::Vector{UInt16})
-        if length(data) < 1 || data[end] != 0
-            throw(UnicodeError(UTF_ERR_NULL_16_TERMINATE, 0, 0))
+        immutable UTF8String <: AbstractString
+            data::Vector{UInt8}
```
Comment (ararslan) [inline src/LegacyStrings.jl]: Out of curiosity, why `Vector{UInt8}` here and `Array{UInt8,1}` for ASCII?

**Comment by nalimilan** [comment]
Let's maybe wait for https://github.com/JuliaLang/julia/pull/16590 to be merged, so that I can make the version condition more precise?

File: src/LegacyStrings.jl (hunk 2 above)
Comment (nalimilan) [inline src/LegacyStrings.jl]: Note I didn't change the code at all. I guess that's just for historical reasons.

Comment (ararslan) [inline src/LegacyStrings.jl]: Oh okay. ¯\\\_(ツ)\_/¯

File: src/LegacyStrings.jl [hunk 3]
```diff
@@ -40,40 +40,43 @@ import Base:
...
     write
 
-immutable ASCIIString <: DirectIndexString
-    data::Array{UInt8,1}
-end
-
-immutable UTF8String <: AbstractString
-    data::Vector{UInt8}
-end
+    if VERSION >= v"0.5.0-"
```
Comment (tkelman) [inline src/LegacyStrings.jl]: there's also this one

**Comment by nalimilan** [comment]
Merging since JuliaLang/julia#16590 seems to be waiting for improvements in LegacyStrings.

**Pull request merged**